var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -29,11 +29,12 @@ class M4(AutotoolsPackage):
     """GNU M4 is an implementation of the traditional Unix macro processor."""
 
     homepage = "https://www.gnu.org/software/m4/m4.html"
-    url      = "https://ftp.gnu.org/gnu/m4/m4-1.4.17.tar.gz"
+    url      = "https://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz"
 
     version('1.4.18', 'a077779db287adf4e12a035029002d28')
     version('1.4.17', 'a5e9954b1dae036762f7b13673a2cf76')
 
+    patch('gnulib-pgi.patch', when='@1.4.18')
```

Comment:
I could if you want, but according to the developer who wrote the patch:

> A half-dozen or so of the thousands of Gnulib warnings were actual portability bugs in Gnulib, which made the exercise worth doing I suppose.

From what he said, "actual portability bugs" are worth fixing for every compiler, even if it was only a deal-breaker for PGI.